### PR TITLE
add `additional` and `default` keywords to the TextMate grammar

### PR DESCRIPTION
**Pull request by ginsbach**
This PR updates the TextMate grammar to contain the recently added `additional` and `default` keywords. This will improve syntax highlighting.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by alexet — APPROVED** [review]

File: extensions/ql-vscode/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## [UNRELEASED]
 
+- Implement syntax highlighting for the `additional` keyword.
```
Comment (charisk) [inline extensions/ql-vscode/CHANGELOG.md:5]: Do you mind including the PR link to the CHANGELOG entries? E.g.

```suggestion
- Implement syntax highlighting for the `additional` keyword. [#2609](https://github.com/github/vscode-codeql/pull/2609)
```

Comment (aeisenberg) [inline extensions/ql-vscode/CHANGELOG.md:5]: Please link to this PR like other changelog entries have.

File: syntaxes/ql.tmLanguage.json
Comment (aeisenberg) [inline syntaxes/ql.tmLanguage.json:1]: Is this file auto-generated from the yaml file? Any other semantic changes in it?

File: extensions/ql-vscode/CHANGELOG.md (hunk 1 above)
Comment (ginsbach) [reply extensions/ql-vscode/CHANGELOG.md]: I have done that in commit 4.

Comment (ginsbach) [reply extensions/ql-vscode/CHANGELOG.md]: I have done that in commit 4.

Comment (charisk) [reply extensions/ql-vscode/CHANGELOG.md]: Thanks!

**Review by aeisenberg — APPROVED** [review]

**Review by aeisenberg — APPROVED** [review]

**Pull request merged**